### PR TITLE
Mock smartstore enhancements / fixes

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: test/MockSmartStore.js [hunk 1]
```diff
@@ -314,7 +314,13 @@ var MockSmartStore = (function(window) {
                     name: "Count of soup items", 
                     example: "SELECT count(*) FROM {soupName}", 
                     pattern: /SELECT count\(\*\) FROM {(.*)}/i, 
-                    processor: this._smartQuerySoupCount 
+                    processor: this._smartQuerySoupCount
+                },
+                { 
+                    name: "Comparing soup item to integer with optional order by",
```
Comment (wmathurin) [inline test/MockSmartStore.js]: That work was mostly done in https://github.com/forcedotcom/SalesforceMobileSDK-Shared/issues/238

File: test/MockSmartStore.js [hunk 2]
```diff
@@ -314,7 +314,13 @@ var MockSmartStore = (function(window) {
                     name: "Count of soup items", 
                     example: "SELECT count(*) FROM {soupName}", 
                     pattern: /SELECT count\(\*\) FROM {(.*)}/i, 
-                    processor: this._smartQuerySoupCount 
+                    processor: this._smartQuerySoupCount
+                },
+                { 
+                    name: "Comparing soup item to integer with optional order by", 
+                    example: "SELECT {soupName:_soup} FROM {soupName} WHERE {soupName:whereField} > 123456 ORDER BY LOWER({soupName:orderByField})", 
+                    pattern: /SELECT {(.*):_soup} FROM {(.*)} WHERE {(.*):(.*)} ([!=<>]+) ([0-9]+)(?: ORDER BY LOWER\({(.*):(.*)}\))?/i,
```
Comment (wmathurin) [inline test/MockSmartStore.js]: Supporting more comparisons types than the original pull request
Keeping the regexp for the comparator simple - the code will only accept !=,=,<,<=,>=

Comment (wmathurin) [inline test/MockSmartStore.js]: Also added support for optional order by (was not in original pull request)

**Comment by bhariharan** [comment]
:+1:

File: test/MockSmartStore.js [hunk 3]
```diff
@@ -518,7 +572,17 @@ var MockSmartStore = (function(window) {
 
             // Since original cursor from smarstore doesn't contain querySpec and soupName, 
             // remove them here too before returning cursor to the user.
-            return _.omit(cursor, 'soupName', 'querySpec');
+            return this.omit(cursor, 'soupName', 'querySpec');
+        },
+
+        omit: function(obj, varNames) {
```
Comment (wmathurin) [inline test/MockSmartStore.js]: So that we don't require underscore to use mock smartstore

File: test/MockSmartStore.js [hunk 4]
```diff
@@ -534,12 +598,12 @@ function hookToCordova(cordova, store, globalStore) {
     var SMARTSTORE_SERVICE = "com.salesforce.smartstore";
 
     cordova.interceptExec(SMARTSTORE_SERVICE, "pgGetDatabaseSize", function (successCB, errorCB, args) {
-        var targetStore = args[0].isGlobalStore ? store : globalStore;
```
Comment (wmathurin) [inline test/MockSmartStore.js]: I got them flipped yesterday

File: test/SFSmartStoreTestSuite.js [hunk 5]
```diff
@@ -931,6 +937,90 @@ SmartStoreTestSuite.prototype.testSmartQueryWithCount  = function() {
 };
 
 /**
+ * TEST smart query with compare on integer field
+ */
+SmartStoreTestSuite.prototype.testSmartQueryWithIntegerCompare  = function() {
```
Comment (wmathurin) [inline test/SFSmartStoreTestSuite.js]: New test

**Comment by khawkins** [comment]
:+1:

**Pull request merged**